### PR TITLE
feat: Partial lambda Function  Fetch

### LINKED ISSUE #1191: Lambda table not been populated when using ViewOnlyAccess
### Describe the bug

Although "Lambda:List*" is present, no error is received and Lambda tables are completely empty when running Cloudquery with ViewOnlyAccess set of permissions.

### Expected Behavior

Populate the tables with the data we can get using the "lambda:List*" permissions present on the ViewOnlyAccess

### Steps to Reproduce

Execute Cloudquery with ViewOnlyAccess set of permissions.

### Possible Solution

_No response_

### Provider and CloudQuery version

AWS

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by bbernays**
#### Summary

Here is an experimental change, where we can populate some data for Lambda Functions as long as the user has access to `lambda:List*` 

Pros:
- As long as user has access to List they can get some data in the table, they will miss information `types.FunctionCodeLocation`, `types.Concurrency` and `Tags`

Cons:
- Includes information about function that could have been deleted
- Continues to try to get details as we cannot be sure if a user doesn't have access to get a specific function or all functions

---

Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [ ] Run `go fmt` to format your code 🖊
- [ ] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [ ] Update or add tests. Learn more about testing [here](https://docs.cloudquery.io/docs/developers/sdk/testing) 🧪
- [ ] Update the docs by running `go run ./docs/docs.go` and committing the changes 📃
- [ ] If adding a new resource, add [relevant Terraform files](../terraform) in a separate PR 📂
- [ ] Ensure the status checks below are successful ✅

### CODE REVIEW

File: client/helpers.go [hunk 1]
```diff
@@ -301,6 +307,29 @@ func isNotFoundError(err error) bool {
 	return false
 }
 
+// IsAccessDeniedError checks if api error should be classified as a permissions issue
+func (c *Client) IsAccessDeniedError(err error) bool {
+	if isAccessDeniedError(err) {
+		c.logger.Warn("API returned \"NotFound\" error ignoring it...", "error", err)
```
Comment (candiduslynx) [inline client/helpers.go:313]: NotFound -> AccessDenied

File: client/helpers.go [hunk 2]
```diff
@@ -301,6 +307,29 @@ func isNotFoundError(err error) bool {
...
+	return false
+}
+
+func isAccessDeniedError(err error) bool {
+	var ae smithy.APIError
+	if !errors.As(err, &ae) {
+		return false
+	}
+	errorCode := ae.ErrorCode()
+	for _, s := range accessDeniedErrorStrings {
```
Comment (candiduslynx) [inline client/helpers.go:325]: maybe do a map instead?
```go
var accessDeniedErrorStrings = map[string]struct{}{
	"AuthorizationError": struct{}{},
	"AccessDenied": struct{}{},
	"AccessDeniedException": struct{}{},
	"InsufficientPrivilegesException": struct{}{},
	"UnauthorizedOperation": struct{}{},
}

func isAccessDeniedError(err error) bool {
	var ae smithy.APIError
	if !errors.As(err, &ae) {
		return false
	}
	errorCode := ae.ErrorCode()
	_, ok := accessDeniedErrorStrings[errorCode]
	return ok
}
```

**Review by candiduslynx — CHANGES_REQUESTED** [review]

**Review by disq — APPROVED** [review]

File: client/helpers.go [hunk 3]
```diff
@@ -301,6 +307,24 @@ func isNotFoundError(err error) bool {
 	return false
 }
 
+// IsAccessDeniedError checks if api error should be classified as a permissions issue
+func (c *Client) IsAccessDeniedError(err error) bool {
+	if isAccessDeniedError(err) {
+		c.logger.Warn("API returned \"Access Denied\" error ignoring it...", "error", err)
```
Comment (candiduslynx) [inline client/helpers.go:313]: Access Denied -> AccessDenied ?

**Review by candiduslynx — APPROVED** [review]

File: client/helpers.go (hunk 3 above)
Comment (disq) [reply client/helpers.go:313]: Actually we don't know if it's `Access Denied` or some other error in the `accessDeniedErrorStrings` map, so log line not that accurate.

**Review by yevgenypats — APPROVED** [review]

File: client/helpers.go (hunk 3 above)
Comment (bbernays) [reply client/helpers.go]: Updated the message

**Pull request merged**